docs/concepts/what-is-a-local-activity.md

```diff
@@ -13,15 +13,23 @@ A Local Activity is an [Activity Execution](/concepts/what-is-an-activity-execut
 Some Activity Executions are very short-living and do not need the queuing semantic, flow control, rate limiting, and routing capabilities.
 For this case, Temporal supports the Local Activity feature.
 
-The main benefit of Local Activities is that they use less Temporal service resources (e.g. lower state transitions) and have much lower latency overhead (because no need to roundtrip to the Cluster) compared to normal Activity Executions.
+The main benefit of Local Activities is that they use less Temporal Cluster resources (for example, fewer History events) and have much lower latency overhead (because no need to roundtrip to the Cluster) compared to normal Activity Executions.
 However, Local Activities are subject to shorter durations and a lack of rate limiting.
 
 Consider using Local Activities for functions that are the following:
 
 - can be implemented in the same binary as the Workflow that calls them.
 - do not require global rate limiting.
 - do not require routing to a specific Worker or Worker pool.
-- no longer than a few seconds, inclusive of retries (shorter than the Workflow Task Timeout, which is 10 seconds by default).
+- no longer than a few seconds, inclusive of retries.
+
+If it takes longer than 80% of the Workflow Task Timeout (which is 10 seconds by default), the Worker will ask the Cluster to create a new Workflow Task to extend the "lease" for processing the Local Activity.
+The Worker will continue doing so until the Local Activity has completed.
+This is called Workflow Task Heartbeating.
+The drawbacks of long-running Local Activities are:
+
+- Each new Workflow Task results in 3 more Events in History.
+- New Commands created by the Workflow concurrently with the Local Activity will not be sent to the Cluster until either the Local Activity completes or the next Worker Task Heartbeat.
```

Comment:
Didn’t read the whole doc so it may be mentioned elsewhere. Another drawback is that while a local activity is running the workflow will not get notified of new events like signals and completions until a workflow task heartbeat is emitted.

Reply:
```suggestion
- The Workflow won't get notified of new events like Signals and completions until the next Workflow Task Heartbeat.
- New Commands created by the Workflow concurrently with the Local Activity will not be sent to the Cluster until either the Local Activity completes or the next Workflow Task Heartbeat.
```